pkg/jobconfig/files.go

```diff
@@ -485,20 +482,23 @@ func mergePresubmits(old, new *prowconfig.Presubmit) prowconfig.Presubmit {
 		return ""
 	}()
 
-	if new.RunIfChanged != "" || new.SkipIfOnlyChanged != "" || new.Annotations["pipeline_run_if_changed"] != "" {
-		merged.RunIfChanged = new.RunIfChanged
-		merged.SkipIfOnlyChanged = new.SkipIfOnlyChanged
-		merged.AlwaysRun = new.AlwaysRun
-	}
-
-	// TODO(muller): Special case images jobs for now. Some repos are marking
-	// images jobs as optional for which we do not have syntax in ci-operator (should we?).
-	// Tolerate manual changes for these jobs for now
+   // TODO(muller): Special case images jobs for now. Some repos are marking
+   // images jobs as optional for which we do not have syntax in ci-operator (should we?).
+   // Tolerate manual changes for these jobs for now
+   // TODO(hector): Image jobs should keep considering always_run for now.
 	if strings.HasSuffix(merged.Name, "-images") {
 		merged.Optional = old.Optional
 		if new.Optional {
 			merged.Optional = new.Optional
 		}
+		merged.RunIfChanged = old.RunIfChanged
```

Comment:
instead of your section please try:
```
if old.AlwaysRun {
  merged.AlwaysRun = old.AlwaysRun
}
```


For now it should be enough, let me know if that worked for you.